lib/elixir/lib/code.ex

```diff
@@ -1286,6 +1286,18 @@ defmodule Code do
     end
   end
 
+  @doc """
+  Checks if the current process can ensure modules are compiled.
+
+  Returns true if the current process is running inside the compiler and
+  calling `ensure_compiled/1` will halt the current process until the given
+  module is compiled. Returns false when the current process is not capable
+  of compiling code and `ensure_compiled/1` simply falls back to `ensure_loaded/2`.
+  """
```

Comment:
```suggestion
  """
  @doc since: "1.11.0"
  @spec ensure_compilable? :: boolean
```

Reply:
What do you think about `can_compile?`, `compiling?`, or `compiler_available?` (or similar) instead of `ensure_compilable?` ?

Reply:
So I thought about all of those. The issue is that in Elixir I can compile code at any time in different ways. I can call, for example, `Code.compile_string`. But all of those other mechanisms do not perform "module resolution". What we are interested here is the module resolution aspect, hence the name.

Reply:
The name is still confusing imo. `ensure_compilabe?/0` does not take any arguments, so what is compilable? Is it the current process? But processes cannot be compiled...

How about `inside_compiler?`, `current_compiling?`, `compiler_process?(self())`, `can_resolve_modules?`, or `module_resolution_available?`.

Reply:
What about `can_ensure_compile?`?

Reply:
I still think that is confusing :). Usually when we use the word "ensure" in the stdlib it means: "unless THING is already done, do THING".

Examples:
`Application.ensure_started`: unless application is already started, start application.
`Code.ensure_loaded`: unless module is already loaded, load module.

But in this case there is no action, it is only a check, so I don't think "ensure" makes sense.

Reply:
So I want to ask "will ensure_compile run or will it fallback to ensure_loaded?" or "can I wait until modules are compiled?". Because this behaviour is provided by "ensure_compiled",  I thought  it would be easier to link to it. Otherwise, I think `can_await_for_modules_compilation?` is the best I can do. :D

To recap, The issue with "inside_compiler?", "compiler_process?" and all others is that it doesn't cover this nuance. As I mentioned above, it is totally possible to be inside the compiler but without any module resolution.

Reply:
Would `inside_parallel_compiler?` be correct?

Reply:
I am :-1: because if we want to provide module awaiting for something besides the parallel compiler, the function name either becomes incorrect or it doesn't fully provide all the information we need.

Reply:
Right, that makes sense.

Reply:
Despite the length, the suggestion `can_await_for_modules_compilation?` seems the closest so far. Perhaps it can be reduced to something shorter such as `can_await_compilation?`?

Reply:
I think the module bit is important. Maybe `can_await_module_compilation?`?

Reply:
That seems appropriate.